lua/terrortown/menus/gamemode/base_gamemodemenu.lua

```diff
@@ -102,6 +107,56 @@ function CLGAMEMODEMENU:AddSubmenu(submenu)
 	self.submenus[#self.submenus + 1] = submenu
 end
 
+---
+-- Enables searchbar for the menu.
+-- @param bool active
+-- @realm client
+function CLGAMEMODEMENU:EnableSearchbar(active)
+	self.searchbar = active
+end
+
+---
+-- Checks if the menu has a searchbar enabled.
+-- @return boolean Returns true if searchbar is set
+-- @realm client
+function CLGAMEMODEMENU:HasSearchbar()
+	return self.searchbar
+end
+
+---
+-- Sets the used searchfunction.
+-- Parameters for that function are submenuClasses and the searchText
+-- @param function searchFunction
+-- @realm client
+function CLGAMEMODEMENU:SetSearchFunction(searchFunction)
+	if not isfunction(searchFunction) then return end
+
+	self.searchFunction = searchFunction
+end
```

Comment:
I'm not sure I like this setter here, why not define the function directly on the class? Is there a reason I'n not seeing right now?